contracts/FlowIDTableStaking.cdc

```diff
@@ -886,26 +902,90 @@ pub contract FlowIDTableStaking {
+                let nodeRecord = FlowIDTableStaking.borrowNodeRecord(nodeID)
+
+                // Each node's rewards can be decreased by a different percentage
+                // Its delegator's rewards are also decreased by the same percentage
+                let rewardDecreasePercentage = nonOperationalNodes[nodeID]!
+
+                // Calculate the normal reward amount, then the rewards left after the decrease
+                var nodeRewardAmount = nodeRecord.tokensStaked.balance * totalRewardScale
+                var nodeRewardsLeft = nodeRewardAmount * rewardDecreasePercentage
+
+                // Add the remaining to the total number of rewards withheld
+                sumRewardsWithheld = sumRewardsWithheld + (nodeRewardAmount - nodeRewardsLeft)
+
+                let rewardsBreakdown = FlowIDTableStaking.RewardsBreakdown(nodeID: nodeID)
+
+                // Iterate through all the withheld node's delegators
+                // and calculate their decreased rewards as well
+                for delegator in nodeRecord.delegators.keys {
+                    let delRecord = nodeRecord.borrowDelegatorRecord(delegator)
+
+                    // Calculate the amount of tokens that this delegator receives
+                    // minus the decrease from the non-operational node
+                    var delegatorRewardAmount = delRecord.tokensStaked.balance * totalRewardScale
+                    var delegatorRewardsLeft = delegatorRewardAmount * rewardDecreasePercentage
+
+                    // Add the withheld rewards to the total sum
+                    sumRewardsWithheld = sumRewardsWithheld + (delegatorRewardAmount - delegatorRewardsLeft)
+
+                    if delegatorRewardsLeft == 0.0 { continue }
+
+                    // take the node operator's cut
+                    if (delegatorRewardsLeft * FlowIDTableStaking.nodeDelegatingRewardCut) > 0.0 {
+
+                        let nodeCutAmount = delegatorRewardsLeft * FlowIDTableStaking.nodeDelegatingRewardCut
+
+                        nodeRewardsLeft = nodeRewardsLeft + nodeCutAmount
+
+                        delegatorRewardsLeft = delegatorRewardsLeft - nodeCutAmount
+                    }
+                    rewardsBreakdown.delegatorRewards[delegator] = delegatorRewardsLeft
+                }
+
+                rewardsBreakdown.nodeRewards = nodeRewardsLeft
+                rewardsBreakdownArray.append(rewardsBreakdown)
+            }
+
+            var withheldRewardsScale = sumRewardsWithheld / totalStaked
+
             /// iterate through all the nodes to pay
             for nodeID in allNodeIDs {
+                if nonOperationalNodes[nodeID] != nil { continue }
+
                 let nodeRecord = FlowIDTableStaking.borrowNodeRecord(nodeID)
 
-                var nodeRewardAmount = nodeRecord.tokensStaked.balance * totalRewardScale
+                var nodeRewardAmount = nodeRecord.tokensStaked.balance * (totalRewardScale + withheldRewardsScale)
```

Comment:
`(totalRewardScale + withheldRewardsScale)` is calculated again on line 1000 below. Is it worth calculating it once and assigning it to a local variable, for clarity and ease of future change?

Reply:
good suggestion! I'm gonna change that